include/seqan3/alignment/matrix/detail/aligned_sequence_builder.hpp

```diff
@@ -172,15 +147,21 @@ class aligned_sequence_builder
             traces.emplace_back(last_dir, span);
         }
 
-        matrix_coordinate path_begin = trace_it.coordinate();
+        res.begin = trace_it.coordinate();
 
-        fst_aligned_t fst_aligned{fst_rng | views::slice(path_begin.col, path_end.col)};
-        sec_aligned_t sec_aligned{sec_rng | views::slice(path_begin.row, path_end.row)};
+        if (band_col_offset.has_value())
+        {
+            res.begin.row += res.begin.col - band_col_offset.value();
+            res.end.row += res.end.col - band_col_offset.value();
+        }
```

Comment:
yes, I agree.  When adapting it for the banded case I realised that this is needs to be changed. Back then I thought, I do it later. Just get finished with the PR. 
But now I really reconsider it and will change it before. 

Just as a note, this means to augment the two_dimensional matrix for banded cases. Because this is the only place that knows the mapping between real matrix and the underlying buffer.

Reply:
Ok, I fixed it in another PR. This one will be rebased on that.